### PR TITLE
improve psum_scatter docstring (formatting and content)

### PR DESCRIPTION
**Pull request by mattjj**

### CODE REVIEW

File: jax/_src/lax/parallel.py [hunk 1]
```diff
@@ -1478,26 +1478,31 @@ def psum_scatter(x, axis_name, *, scatter_dimension=0, axis_index_groups=None, t
```
Comment (hawkinsp) [inline jax/_src/lax/parallel.py:1478]: I have to say this description isn't good enough to communicate the intuition, and it's probably better to get the intuition before reading the details...

**Review by hawkinsp — APPROVED** [review]

File: jax/_src/lax/parallel.py (hunk 1 above)
Comment (mattjj) [reply jax/_src/lax/parallel.py:1478]: You're right. Maybe like "Compute a ``psum``, but retain only part of the result on each device."? Or something more precise?

Comment (mattjj) [reply jax/_src/lax/parallel.py:1478]: For precision, we could add that `psum_scatter(x, axis_name, scatter_dimension=0, tiled=False)` produces the same value as `psum(x, axis_name)[jax.lax.axis_index(axis_name)]`, but more efficient. But that doesn't fit in the first line of the docstring.

Comment (mattjj) [reply jax/_src/lax/parallel.py:1478]: image

**Pull request merged**